services/distribution/src/test/java/app/coronawarn/server/services/distribution/statistics/KeyFigureCardFactoryTest.java

```diff
@@ -261,4 +273,163 @@ private void assertKeyFigure(KeyFigure result, double value, Rank rank, Trend tr
         .containsExactly(value, rank, trend, trendSemantic, decimals);
   }
 
+  @Nested
+  @ExtendWith(SpringExtension.class)
+  @EnableConfigurationProperties(value = DistributionServiceConfig.class)
+  @ContextConfiguration(classes = {KeyFigureCardFactory.class},
+      initializers = ConfigDataApplicationContextInitializer.class)
+  class VaccinationDosesCardFactoryTest {
+
+    @Test
+    void testCardHasCorrectKeyFigures() {
+      var result = figureCardFactory.createKeyFigureCard(statisticsJsonStringObject, 7);
+
+      assertKeyFigure(result.getKeyFigures(0), 843978, Rank.PRIMARY, Trend.UNSPECIFIED_TREND,
+          TrendSemantic.UNSPECIFIED_TREND_SEMANTIC, 0);
+      assertKeyFigure(result.getKeyFigures(1), 656704.8571428572, Rank.SECONDARY, Trend.INCREASING,
+          TrendSemantic.POSITIVE, 0);
+      assertKeyFigure(result.getKeyFigures(2), 41517849, Rank.TERTIARY, Trend.UNSPECIFIED_TREND,
+          TrendSemantic.UNSPECIFIED_TREND_SEMANTIC, 0);
+    }
+
+    @Test
+    void testAdministeredDosesDecreasing() {
+      statisticsJsonStringObject.setAdministeredDoses7daysAvgTrend5percent(-1);
+      var result = figureCardFactory.createKeyFigureCard(statisticsJsonStringObject,
+          VACCINATION_DOSES_CARD);
+      assertThat(result.getKeyFigures(1))
+          .extracting(KeyFigure::getTrend, KeyFigure::getTrendSemantic)
+          .containsExactly(Trend.DECREASING, TrendSemantic.NEGATIVE);
+    }
+
+    @Test
+    void testAdministeredDosesIncreasing() {
+      statisticsJsonStringObject.setAdministeredDoses7daysAvgTrend5percent(1);
+      var result = figureCardFactory.createKeyFigureCard(statisticsJsonStringObject,
+          VACCINATION_DOSES_CARD);
+      assertThat(result.getKeyFigures(1))
+          .extracting(KeyFigure::getTrend, KeyFigure::getTrendSemantic)
+          .containsExactly(Trend.INCREASING, TrendSemantic.POSITIVE);
+    }
+
+    @Test
+    void shouldThrowAnExceptionIfMandatoryPropertyIsEqualToZero() {
```

Comment:
These could maybe collapsed into one ParameterizedTest ( testing if an Exception is thrown)